data/virt_autotest/virt_logs_collector.sh

```diff
@@ -0,0 +1,427 @@
+      }
+
+      if { \${ret_result} == 1 && \${extra_logs} != "" } {
+         expect -nocase "login: $" {puts "Will try to collect \${extra_logs} from \${guest_domain} again\r"}
+      }
+
+      if { \${extra_logs} == "" } {
+         expect -nocase "login: $" {puts "Nothing to collect from \${guest_domain}. Please specify something.\r"}
+      }
+
+      set retry_times [expr \${retry_times}-1]
+      close -i \${spawn_id}
+}
+exit \${ret_result}
+EOF
+
+        echo -e "${expfile} to be executed is as below:"
+        cat ${expfile}
+  
+	for procid in `ps aux | grep  ${expfile} | grep -v "grep" | awk '{print $2}'`;do
+            kill -9 ${procid}
+	done
+	for procid in `ps aux | grep "virsh console" | grep -v "grep" | awk '{print $2}'`;do
+	    kill -9 ${procid}
+	done
+	echo -e "expect ${expfile} "${hypervisor}" "${guest_domain}" "${guest_transformed}" "${logs_folder}" "${extra_logs}""
+	expect ${expfile} "${hypervisor}" "${guest_domain}" "${guest_transformed}" "${logs_folder}" "${extra_logs}"
+        if [[ $? == 0 ]];then
+           echo -e "${expfile} returned with success. Successfully collected ${extra_logs} via ${guest_domain} console."
+           rm -f ${expfile}
+	   return 0
+        else
+           echo -e "${expfile} returned with failure. Failed to collect ${extra_logs} via ${guest_domain} console."
+           rm -f ${expfile}
+           return 1
+	fi
+}
+
+#Collect supportconfig from both host and guest. Will resort to guest virsh console if collecting from ssh failed.
+function collect_supportconfig() {
+	local logs_folder=$1
+	local target_type=$2
+	local target_ipaddr=$3
+        local target_domain=$4
+	local target_user=""
+	local target_pass=""
+	local sshpass_ssh_cmd=""
+
+	if [[ ${target_type} == "guest" ]];then
+           target_user="root"
+           target_pass="novell"
+	   sshpass_ssh_cmd="sshpass -p ${target_pass} ssh ${target_user}@${target_ipaddr}"
+	fi	
+
+	if [[ ${target_domain} != "" ]];then
+	   local target_transformed=${target_domain//./_}
+        else 
+	   local target_transformed=`hostname`
+	   target_transformed=${target_transformed//./_}
+	fi
+
+	local ret_result=128
+	local retry_times=0
+	while [[ ${retry_times} -lt 3 ]] && [[ ${ret_result} -ne 0 ]];
+	do	
+    	      local time_stamp=`date '+%Y%m%d%H%M%S'`
+	      ${sshpass_ssh_cmd} rm -f -r ${logs_folder}
+              echo -e "${sshpass_ssh_cmd} supportconfig -t ${logs_folder} -B ${target_type}_${target_transformed}_supportconfig_${time_stamp}"
+	      ${sshpass_ssh_cmd} supportconfig -t ${logs_folder} -B ${target_type}_${target_transformed}_supportconfig_${time_stamp}
+	      ret_result=$?
+	      if [[ ${ret_result} -eq 0 ]];then
+                 echo -e "Successfully collected supportconfig from ${target_type} ${target_domain} via ssh."
+		 break
+	      fi
+	      retry_times=$((${retry_times}+1))
+	done
+
+	if [[ ${target_type} == "guest" ]] && [[ ${ret_result} -ne 0 ]];then
```

Comment:
The indent of the code block within this if is not 4 blanks and some not aligned.